src/nnsight/tracing/Node.py

```diff
@@ -105,7 +105,7 @@ def preprocess_node(node: Union[Node, Proxy]):
 
             if self.attached() and self.graph.id != node.graph.id:
 
-                node = protocols.BridgeProtocol.add(node).node
+                node = protocols.BridgeProtocol.bridge_node(node, self.graph.id)
```

Comment:
Is bridge_node replacing add? If so why?

Reply:
Yes it is. Because bridging a node should not necessarily involve adding a new `BridgeProtocol` node.  We would like it to use an existing one if applicable (to maintain the  invariant of one unique Bridge node per external proxy).

`.bridge_node()` calls `.add()` if a `BridgeProtocol` for the corresponding proxy does not exist yet.